receiver/dotnetdiagnosticsreceiver/dotnet/request.go

```diff
@@ -0,0 +1,240 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dotnet
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"math"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dotnetdiagnosticsreceiver/network"
+)
+
+// RequestWriter submits the initial request to the dotnet diagnostics backend,
+// after which it starts sending us metrics at the specified interval.
+// ByteBuffer is swappable for testing.
+type RequestWriter struct {
+	buf         network.ByteBuffer
+	w           io.Writer
+	intervalSec int
+	names       []string
```

Comment:
What are the names? Please add comment.

Reply:
Sure, I added a comment and changed the name of the variable to make it clearer.